File: Segundo ano/2 semestre/Projeto e Análise de Algoritmos/fibonacciRecursivo.c
#include <stdio.h>
#include <sys/time.h>
#include <time.h>

unsigned long long int fibonacciRecursivo(int N) {
    if (N <= 1)
        return N;
    else
        return fibonacciRecursivo(N - 1) + fibonacciRecursivo(N - 2);
}

int main() {
    int N;
    unsigned long long int resultado;
    N = 10;
    //struct timeval inicio, fim;
    //double tempo;
    clock_t inicio, fim;
    long double tempo;

    inicio = clock();
    resultado = fibonacciRecursivo(N);
    fim = clock();

    tempo = ((long double)(fim - inicio))/CLOCKS_PER_SEC;
    printf("N: %d\nResultado: %llu\nTempo: %Lf segundos", N, resultado, tempo);
    
    return 0;
}

File: Segundo ano/2 semestre/Projeto e Análise de Algoritmos/fibonacciRecursivo.py
import time

def fibonacciRecursivo(n):
  if n <= 1:
    return n
  else:
    return fibonacciRecursivo(n-1) + fibonacciRecursivo(n-2)

inicio = time.time()
resultado = fibonacciRecursivo(50)
fim = time.time()
print(fim-inicio, resultado)